### PR TITLE
Use the Application-Defined User Class instead of `Spree::User`

### PR DESCRIPTION
**Pull request by pmn4**
not an issue in most cases, but seems like the right thing to do.

_(there are some failing tests, but they seem to be failing on the parent commit as well)_

### CODE REVIEW

**Review by jarednorman — CHANGES_REQUESTED** [review]
Can we revert the changes to the specs so they still include the explicit class? Tests should avoid "computed" values to reduce the risk of false positives, `Spree.user_class` should always be `Spree::User` in our test suite.

**Comment by pmn4** [comment]
💯 great point.

File: app/controllers/spree/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def omniauth_callback
       flash[:notice] = I18n.t('devise.sessions.signed_in')
       redirect_back_or_default(account_url)
     else
-      user = Spree::User.find_by_email(auth_hash['info']['email']) || Spree::User.new
+      user = Spree.user_class.find_by_email(auth_hash['info']['email']) || Spree.user_class.new
```
Comment (pmn4) [inline app/controllers/spree/omniauth_callbacks_controller.rb:37]: @jarednorman I am having second thoughts about the first change on this line too. Is it safe to assume that the class defined by `Spree.user_class` inherits from `Spree::User`? If so, I am going to revert to:
``` ruby
Spree::User.find_by_email(auth_hash['info']['email']) || Spree.user_class.new
```

Comment (pmn4) [reply app/controllers/spree/omniauth_callbacks_controller.rb:37]: nevermind. `Spree::LegacyUser` doesn't even inherit from `Spree::User`

**Pull request merged**